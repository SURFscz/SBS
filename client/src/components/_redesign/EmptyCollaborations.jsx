import React from "react";
import {ReactComponent as TreeSwing} from "../../images/tree_swing.svg";

import "./Collaborations.scss";
import I18n from "../../locale/I18n";
import Button from "../button/Button";
import {isUserAllowed, ROLES} from "../../utils/UserRole";
import {isEmpty} from "../../utils/Utils";


export default class EmptyCollaborations extends React.PureComponent {

    render() {
        const {user} = this.props;
        const isOrgManager = isUserAllowed(ROLES.ORG_MANAGER, user);
        const organisationsFromUserSchacHome = user.organisations_from_user_schac_home || [];
        const createFromSchacHome = organisationsFromUserSchacHome
            .some(org => org.collaboration_creation_allowed || org.collaboration_creation_allowed_entitlement)

        const mayCreateCollaborations = isOrgManager || createFromSchacHome;
        const organisationsQueryParam = isEmpty(organisationsFromUserSchacHome) ? "" : `?organisationId=${organisationsFromUserSchacHome.map(org => org.id).join(",")}`;
        const newLabel = I18n.t(`models.${mayCreateCollaborations ? "collaborations" : "memberCollaborations"}.new`);
        const newPath = `/new-collaboration${organisationsQueryParam}`;
        const info = isOrgManager ? I18n.t("models.collaborations.noCollaborations"):
            I18n.t(`models.collaborations.noCollaborations${mayCreateCollaborations ? "" : "Request"}User`);
        return (
            <div className="no-collaborations">
                <TreeSwing/>
                <p>{info}</p>
                <Button txt={newLabel}
                        onClick={() => {
                            this.props.history.push(newPath)
                        }}/>

            </div>
        )
    }

}
