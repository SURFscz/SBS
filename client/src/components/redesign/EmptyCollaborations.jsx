import React from "react";
import {ReactComponent as TreeSwing} from "../../images/tree_swing.svg";

import "./Collaborations.scss";
import I18n from "../../locale/I18n";
import Button from "../Button";
import {isUserAllowed, ROLES} from "../../utils/UserRole";
import {isEmpty} from "../../utils/Utils";


export default class EmptyCollaborations extends React.PureComponent {

    render() {
        const {user} = this.props;
        const isOrgManager = isUserAllowed(ROLES.ORG_MANAGER, user);
        const organisationFromUserSchacHome = user.organisation_from_user_schac_home || {};
        const mayCreateCollaborations = isOrgManager || organisationFromUserSchacHome.collaboration_creation_allowed ||
            organisationFromUserSchacHome.collaboration_creation_allowed_entitlement;
        const organisationQueryParam = isEmpty(organisationFromUserSchacHome) ? "" : `?organisationId=${organisationFromUserSchacHome.id}`;
        const newLabel = I18n.t(`models.${mayCreateCollaborations ? "collaborations" : "memberCollaborations"}.new`);
        const newPath = `/new-collaboration${organisationQueryParam}`;
        const info = isOrgManager ? I18n.t("models.collaborations.noCollaborations"):
            I18n.t(`models.collaborations.noCollaborations${mayCreateCollaborations ? "" : "Request"}User`);
        return (
            <div className="no-collaborations">
                <TreeSwing/>
                <p>{info}</p>
                <Button txt={newLabel}
                        onClick={() => {
                            this.props.history.push(newPath)
                        }}/>

            </div>
        )
    }

}

