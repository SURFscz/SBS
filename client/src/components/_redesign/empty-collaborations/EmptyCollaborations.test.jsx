import React from "react";
import { render } from "@testing-library/react";
import EmptyCollaborations from "./EmptyCollaborations";

const mockUser = { admin: true };

describe("EmptyCollaborations", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <EmptyCollaborations
                title="No Collaborations"
                description="You have no collaborations yet."
                buttonText="Create Collaboration"
                onButtonClick={() => {}}
                user={mockUser}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
