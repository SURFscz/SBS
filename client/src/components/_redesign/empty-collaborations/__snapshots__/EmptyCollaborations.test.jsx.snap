// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCollaborations matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="no-collaborations"
  >
    <svg>
      tree_swing.svg
    </svg>
    <p>
      There are no collaborations yet within your organisation. Why don't you create the first one?
    </p>
    <button
      class="sds--btn sds--btn--primary    "
      type="button"
    >
      <span
        class="textual"
      >
        Add collaboration
      </span>
    </button>
  </div>
</DocumentFragment>
`;
